main/sig_finder/sig_finder.cc

```diff
@@ -100,7 +97,14 @@ std::optional<SigLoc> findSignature(const core::GlobalState &gs, const core::Sym
             break;
         }
 
-        // If we find anything else, even a blank line, assume that the heuristic
+        // We found a comment between a sig and a method def, look one line higher
+        if (absl::StartsWith(line, commentLineStart)) {
```

Comment:
I guess this heuristic only works when the `#` starts at the line start, with no preceding whitespace.

If we don't want to fix this now, let's either add a comment or a test that documents this this as a known limitation that might be worth fixing later. (This PR is starting to look really good, and this is the sort of small thing that I'm happy to punt on.)

Though, writing up this comment also got me thinking that it might be nice to handle moving documentation comments above the method/sig too:

```ruby
class A
  # My comment on the method
  # @param arg0 Docs for the first argument
  sig {params(arg0: Integer).void}
  def self.foo(arg0); end
end
```

Right now we'd leave the comment behind. This is also something I'm fine punting on for a future change. If you'd like to merge this change before fixing this, I'd love if you could be sure to file an issue to remind us.

Reply:
>I guess this heuristic only works when the # starts at the line start, with no preceding whitespace.

It should work with leading whitespaces too. In the same method the `line` is trimmed so it should be okay. I'll add test for that.
https://github.com/sorbet/sorbet/blob/14bf3533903bd2ea376a323de2657d7b39bf6d20/main/sig_finder/sig_finder.cc#L26

Reply:
https://github.com/sorbet/sorbet/issues/5501